src/components/controls/controls.js

```diff
@@ -0,0 +1,218 @@
+var registerControls = require('../../core/controls').registerControls;
+var THREE = require('../../lib/three');
+
+var MAX_DELTA = 0.2; // ms
+var PI_2 = Math.PI / 2;
+
+/**
+ * Controls component.
+ *
+ * Receives input events from device-specific control components, and applies movement and rotation
+ * to the element accordingly.
+ */
+module.exports.Component = registerControls('controls', {
+  dependencies: ['velocity'],
+
+  schema: {
+    enabled: { default: true },
+
+    flyingEnabled: { default: false },
+
+    position: { default: ['gamepad-controls', 'keyboard-controls', 'touch-controls'] },
```

Comment:
Think you could copy over the look-controls code to hmd-controls, we could test it? If not, yeah I can help.